cmd/boots/tftp.go

```diff
@@ -96,15 +96,17 @@ func (t tftpHandler) ReadFile(c tftp.Conn, filename string) (tftp.ReadCloser, er
 	if !j.AllowPxe() {
 		l.Info("the hardware data for this machine, or lack there of, does not allow it to pxe; allow_pxe: false")
 		span.SetStatus(codes.Error, "allow_pxe is false")
-		span.End()
 
 		return serveFakeReader(l, filename)
 	}
 
-	span.SetStatus(codes.Ok, filename)
-	span.End()
+	// this span is after fetching the hw data so it will be attached to the upstream trace id
+	// we also link the spans here so they can be tied together in the tracing UI
+	ctx, uSpan := tracer.Start(ctx, "TFTP open", trace.WithAttributes(attribute.String("filename", filename)))
```

Comment:
nit: Should be `TFTP Open` to match the case used for DHCP

Reply:
gonna switch these over to the otel semantic recommendations. for this one, "tftp/ServeTFTP" since that's the method the span is wrapping